### PR TITLE
Refactoring with option in parser.y

### PR DESCRIPTION
**Pull request by ydah**
Use new racc's grammar: `?`

### CODE REVIEW

File: spec/lrama/diagnostics_spec.rb [hunk 1]
```diff
@@ -29,6 +29,7 @@
           %token tEQEQ  "=="
 
           %type <i> class /* comment for class */
+          %type string_1 string_2 /* missing `%type` tag */
```
Comment (yui-knk) [inline spec/lrama/diagnostics_spec.rb:32]: What this comment means? I think `%type` is not missed on this grammar file.

Comment (ydah) [reply spec/lrama/diagnostics_spec.rb:32]: What I meant to say there is that the `%type` tag is missing. In other words, I wanted to make it clear that I was testing the lack of `<tag>`.

Comment (yui-knk) [reply spec/lrama/diagnostics_spec.rb:32]: I got it. What do you think about comment like `/* <tag> is omitted */` ?

Comment (ydah) [reply spec/lrama/diagnostics_spec.rb]: Thank you! I updated comment.

**Review by yui-knk — APPROVED** [review]

**Comment by ydah** [comment]
resolve conflict

**Pull request merged**